### PR TITLE
[CD-396] Initial port to public repo

### PR DESCRIPTION
**Pull request by nckswt**
## Summary

Public port of a private plugin. We need to make this public so we can use it on Grafana Cloud.

## TODOs

After merge
- [ ] Submit plugin for review
- [ ] Sign plugin

### CODE REVIEW

File: .config/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ARG grafana_version=latest
+ARG grafana_image=grafana-enterprise
+
+FROM grafana/${grafana_image}:${grafana_version}
+
+# Make it as simple as possible to access the grafana instance for development purposes
```
Comment (TechTeam12) [inline .config/Dockerfile:6]: Are these the right values?

File: .config/webpack/utils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export function hasReadme() {
+  return fs.existsSync(path.resolve(process.cwd(), SOURCE_DIR, 'README.md'));
+}
+
+// Support bundling nested plugins by finding all plugin.json files in src directory
+// then checking for a sibling module.[jt]sx? file.
+export async function getEntries(): Promise<Record<string, string>> {
+  const pluginsJson = await glob('**/src/**/plugin.json', { absolute: true });
```
Comment (TechTeam12) [inline .config/webpack/utils.ts:22]: This is weird... You have a utility function defined on 11 to find the file, then you're globbing for the file?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
```
Comment (TechTeam12) [inline .github/workflows/ci.yml:21]: Is 16 required? It'd be nice to use node matrix, and try 18. Ideally we should be looking at starting our move to 18 soon-ish.

File: src/components/ConfigEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  const onAwsAccessKeyIdChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const jsonData = {
+      ...options.jsonData,
+      awsAccessKeyId: event.target.value,
+    };
+    onOptionsChange({ ...options, jsonData });
+  };
+
+  const onAwsSecretAccessKeyChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const jsonData = {
```
Comment (TechTeam12) [inline src/components/ConfigEditor.tsx:21]: This looks like there is some boiler plate here, but this seems funky. If this is a state, I'm curious how it's being managed and set. Doing the spread on options is going to be weird, you can get a race condition on the frontend.

**Review by TechTeam12 — APPROVED** [review]
Lots of boiler plate I assume, very weird app.

File: .config/Dockerfile (hunk 1 above)
Comment (nckswt) [reply .config/Dockerfile:6]: Yes these are

**Pull request merged**